pkg/controller/eventtype/reconcile.go

```diff
@@ -0,0 +1,215 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventtype
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang/glog"
+	feedsv1alpha1 "github.com/knative/eventing/pkg/apis/feeds/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"strings"
+)
+
+const (
+	finalizerName = controllerAgentName
+)
+
+// Reconcile compares the actual state of a Feed with the desired, and attempts
+// to converge the two. It then updates the Status block of the Feed with
+// its current state.
+// If Reconcile returns a non-nil error, the request will be retried.
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	//TODO use this to store the logger and set a deadline
+	ctx := context.TODO()
+
+	et := &feedsv1alpha1.EventType{}
+	err := r.client.Get(ctx, request.NamespacedName, et)
+
+	// The EventType may have been deleted since it was added to the workqueue. If so
+	// there's nothing to be done.
+	if errors.IsNotFound(err) {
+		glog.Errorf("could not find EventType %v\n", request)
+		return reconcile.Result{}, nil
+	}
+
+	// If the EventType exists but could not be retrieved, then we should retry.
+	if err != nil {
```

Comment:
I would move this to line 48 and inject the IsNotFound test inside of the if err != nil.

could also do 

`if err =  r.client.Get(ctx, request.NamespacedName, et); err != nil {`

Reply:
Lets do that as a follow-up. I tend to see `err`  as a transient variable and something with a special name `errForLater` (not really that name) for special cases to make it clear. This current pattern is confusing...